code/controllers/subsystem/minimap.dm

```diff
@@ -88,15 +88,17 @@ var/datum/subsystem/minimap/SSminimap
 			tile_icon.Scale(TILE_SIZE, TILE_SIZE)
 			// Add the tile to the minimap.
 			minimap.Blend(tile_icon, ICON_OVERLAY, ((tile.x - 1) * TILE_SIZE), ((tile.y - 1) * TILE_SIZE))
+			del(tile_icon)
 		
 		//byond bug, this fixes OOM crashes by flattening and reseting the minimap icon holder every 255 tiles
 		counter--
 		if (counter <= 0)
-			counter = 255
+			counter = 128
 			var/icon/flatten = new /icon()
 			flatten.Insert(minimap, "", SOUTH, 1, 0)
 			del(minimap)
 			minimap = flatten
+			sleep -1
```

Comment:
`sleep -1` is arcane magic from the depths of byond
just don't think about it too hard